File: the-xray-tracer/typescript/lib/the-sqs-flow-stack.ts
import * as cdk from '@aws-cdk/core';
import lambda = require('@aws-cdk/aws-lambda');
import sqs = require('@aws-cdk/aws-sqs');
import { SqsEventSource } from '@aws-cdk/aws-lambda-event-sources';
import sns_sub = require('@aws-cdk/aws-sns-subscriptions');
import sns = require('@aws-cdk/aws-sns');

export interface SQSFlowStackProps extends cdk.StackProps{
    readonly snsTopicARN: string;
}

export class TheSqsFlowStack extends cdk.Stack {
    sqslambda: lambda.Function;
    constructor(scope: cdk.Construct, id: string, props: SQSFlowStackProps) {
        super(scope, id, props);

        /**
         * SQS Flow
         * SQS creation
         */
        const queue = new sqs.Queue(this, 'Queue', {
            visibilityTimeout: cdk.Duration.seconds(300)
        });

        // defines an AWS Lambda resource
        this.sqslambda = new lambda.Function(this, 'sqsLambdaHandler', {
            runtime: lambda.Runtime.NODEJS_12_X,
            code: lambda.Code.fromAsset('lambda-fns'),
            handler: 'sqs.handler',
            environment: {
                SQS_URL: queue.queueUrl
            },
            tracing: lambda.Tracing.ACTIVE
        });
        queue.grantSendMessages(this.sqslambda);

        let topic = sns.Topic.fromTopicArn(this, 'SNSTopic', props.snsTopicARN);
        topic.addSubscription(new sns_sub.LambdaSubscription(this.sqslambda));

        // defines an AWS Lambda resource to pull from our queue
        const sqsSubscribeLambda = new lambda.Function(this, 'sqsSubscribeLambdaHandler', {
            runtime: lambda.Runtime.NODEJS_12_X,
            code: lambda.Code.fromAsset('lambda-fns'),
            handler: 'sqs_subscribe.handler', 
            tracing: lambda.Tracing.ACTIVE
        });
        queue.grantConsumeMessages(sqsSubscribeLambda);
        sqsSubscribeLambda.addEventSource(new SqsEventSource(queue, {}));
    }
}

File: the-xray-tracer/python/the_xray_tracer/the_sqs_flow_stack.py
from aws_cdk import (
    aws_lambda as _lambda,
    aws_lambda_event_sources as lambda_event,
    aws_sns as sns,
    aws_sns_subscriptions as subscriptions,
    aws_sqs as sqs,
    core
)


class TheSqsFlowStack(core.Stack):
    def __init__(self, scope: core.Construct, id: str, sns_topic_arn: str, **kwargs) -> None:
        super().__init__(scope, id, **kwargs)

        # Queue Setup
        sqs_queue = sqs.Queue(self, 'RDSPublishQueue', visibility_timeout=core.Duration.seconds(300))

        sqs_lambda = _lambda.Function(self, "sqsLambdaHandler",
                                      runtime=_lambda.Runtime.NODEJS_12_X,
                                      handler="sqs.handler",
                                      code=_lambda.Code.from_asset("lambda_fns"),
                                      tracing=_lambda.Tracing.ACTIVE,
                                      environment={
                                          "SQS_URL": sqs_queue.queue_url
                                      }
                                      )
        sqs_queue.grant_send_messages(sqs_lambda)

        topic = sns.Topic.from_topic_arn(self, 'SNSTopic', sns_topic_arn)
        topic.add_subscription(subscriptions.LambdaSubscription(sqs_lambda))

        sqs_subscribe_lambda = _lambda.Function(self, "sqsSubscribeLambdaHandler",
                                                runtime=_lambda.Runtime.NODEJS_12_X,
                                                handler="sqs_subscribe.handler",
                                                code=_lambda.Code.from_asset("lambda_fns"),
                                                tracing=_lambda.Tracing.ACTIVE
                                                )
        sqs_queue.grant_consume_messages(sqs_subscribe_lambda)
        sqs_subscribe_lambda.add_event_source(lambda_event.SqsEventSource(sqs_queue))

